src/kvstore/raftex/Host.h

```diff
@@ -53,6 +53,22 @@ class Host final : public std::enable_shared_from_this<Host> {
         stopped_ = true;
     }
 
+    void reset() {
+        std::lock_guard<std::mutex> g(lock_);
+        if (!promise_.isFulfilled()) {
```

Comment:
I am thinking about this logic. 
If we setValue for current promise here, what if the repsonse comes back,  reset the promise again? 
Is it legal?